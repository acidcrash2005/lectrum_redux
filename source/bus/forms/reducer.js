//Core
import { combineForms } from 'react-redux-form';

export const formsReducer = combineForms({
    newTask: {
        message: '',
    },
    filter: {
        tasksFilter: '',
    },
}, 'forms');
