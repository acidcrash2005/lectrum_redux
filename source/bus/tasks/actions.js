// Types
import { types } from './types';

//Async
export const fetchTasksAsync = () => ({
    type: types.FETCH_TASKS_ASYNC,
});

export const createTasksAsync = (message) => ({
    type:    types.CREATE_TASKS_ASYNC,
    payload: message,
});

export const updateTasksAsync = (updateTask) => ({
    type:    types.UPDATE_TASKS_ASYNC,
    payload: updateTask,
});

export const removeTaskAsync = (idTask) => ({
    type:    types.REMOVE_TASKS_ASYNC,
    payload: idTask,
});

export const completeAllTasksAsync = (taskList) => ({
    type:    types.COMPLETE_ALL_TASKS_ASYNC,
    payload: taskList,
});

//Sync
export const fillTasks = (tasks) => {
    return {
        type:    types.FILL_TASKS,
        payload: tasks,
    };
};

export const createTasks = (task) => ({
    type:    types.CREATE_TASK,
    payload: task,
});

export const updateTasks = (updateTask) => ({
    type:    types.UPDATE_TASKS,
    payload: updateTask,
});

export const removeTask = (idTask) => ({
    type:    types.REMOVE_TASKS,
    payload: idTask,
});

export const completeAllTasks = (taskList) => ({
    type:    types.COMPLETE_ALL_TASKS,
    payload: taskList,
});

export const editTaskChange = (newMessage) => ({
    type:    types.EDIT_TASK_CHANGE,
    payload: newMessage,
});

export const editTaskFocus = (taskId) => ({
    type:    types.EDIT_TASK_FOCUS,
    payload: taskId,
});

