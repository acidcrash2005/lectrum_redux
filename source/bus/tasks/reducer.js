//Core
import { Map, List, fromJS } from 'immutable';

// Types
import { types } from './types';

const initialState = Map({
    taskList: List(),
    edit:     Map({
        taskId:     '',
        newMessage: '',
        isEditing:  '',
    }),
});

export const getIndex = (taskList, action) => {
    return taskList.findIndex(
        (task) => task.get('id') === action.payload.id
    );
};

export const tasksReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_TASKS:
            return state.set('taskList', fromJS(action.payload));

        case types.CREATE_TASK:
            return state.updateIn(['taskList'], (taskList) => taskList.unshift(fromJS(action.payload)));

        case types.UPDATE_TASKS:
            return state.updateIn(['taskList'], (taskList) => taskList.map((task, index) => {

                if (index === getIndex(state.get('taskList'), action)) {
                    return task.merge(fromJS(action.payload));
                }

                return task;
            }));

        case types.REMOVE_TASKS:
            return state.updateIn(['taskList'], (taskList) => {
                return taskList.filter((task) => task.get('id') !== action.payload);
            });

        case types.COMPLETE_ALL_TASKS:
            return state.updateIn(['taskList'], (taskList) => {
                return taskList.map((task) => {
                    return task.merge(
                        action.payload.find((item) => item.get('id') === task.get('id'))
                    );
                });
            });

        case types.EDIT_TASK_CHANGE:
            return state.updateIn(['edit', 'newMessage'], () => action.payload);

        case types.EDIT_TASK_FOCUS:
            return state.updateIn(['edit', 'taskId'], () => action.payload);


        default:
            return state;
    }
};
