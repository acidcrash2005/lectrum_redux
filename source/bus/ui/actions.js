// Types
import { types } from './types';

export const startSpinner = () => {
    return {
        type: types.START_FETCHING,
    };
};

export const stopSpinner = () => {
    return {
        type: types.STOP_FETCHING,
    };
};

