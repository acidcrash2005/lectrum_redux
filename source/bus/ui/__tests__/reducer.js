//Reducer
import { uiReducer } from '../reducer';

//Actions
import * as uiActions from '../actions';

describe('auth ui:', () => {
    test('should return state by default', () => {
        expect(uiReducer(void 0, { type: __.testAction })).toMatchSnapshot();
    });

    test('should handle START_FETCHING action', () => {
        expect(uiReducer(void 0, uiActions.startSpinner())).toMatchSnapshot();
    });

    test('should handle STOP_FETCHING action', () => {
        expect(uiReducer(void 0, uiActions.startSpinner())).toMatchSnapshot();
    });

});
