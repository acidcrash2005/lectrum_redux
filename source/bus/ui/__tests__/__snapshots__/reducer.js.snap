// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth ui: should handle START_FETCHING action 1`] = `
Immutable.Map {
  "isFetching": true,
}
`;

exports[`auth ui: should handle STOP_FETCHING action 1`] = `
Immutable.Map {
  "isFetching": true,
}
`;

exports[`auth ui: should return state by default 1`] = `
Immutable.Map {
  "isFetching": false,
}
`;
